packages/ui/src/working-groups/components/WithdrawApplicationButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useCallback } from 'react'
+
+import { useBalance } from '@/accounts/hooks/useBalance'
+import { useGroupLocks } from '@/accounts/hooks/useGroupLocks'
+import { RecoverableLock, RecoverBalanceModalCall } from '@/accounts/modals/RecoverBalance'
+import { ButtonSecondary } from '@/common/components/buttons'
+import { useModal } from '@/common/hooks/useModal'
+import { WorkingGroupApplication } from '@/working-groups/types/WorkingGroupApplication'
+
+interface WithdrawApplicationButtonProps {
+  application: WorkingGroupApplication
+}
+
+export const WithdrawApplicationButton = ({ application }: WithdrawApplicationButtonProps) => {
+  const { showModal } = useModal()
+  const stakingAccountBalance = useBalance(application.stakingAccount)
+  const { recoverable } = useGroupLocks(application.stakingAccount, stakingAccountBalance?.locks ?? [])
+
+  const withdrawApplication = useCallback(() => {
+    const lock = recoverable.find((lock) => lock.type.endsWith('Worker'))
+    if (lock) {
```

Comment:
Similarly to `@/council/components/PastVotes/PastVote/PastVote` it would be nicer to have:
```suggestion
  const hasWGLock = stakingAccountBalance?.locks.some(({ type }) => type.endsWith('Worker'))
  const isWGLockRecoverable = useIsWGLockRecoverable(hasWGLock, address)
  
  const withdrawApplication = useCallback(() => {
    if (!isWGLockRecoverable) return
```